modules/tinymce/src/core/test/ts/browser/UndoManagerTest.ts

```diff
@@ -611,4 +612,68 @@ describe('browser.tinymce.core.UndoManagerTest', () => {
 
     assert.equal(changeEventCounter, 0, 'No events should be detected');
   });
+
+  it('TINY-8641: Dispatch change and set editor dirty if the current content and the undoManager history are incoherent', () => {
+    const editor = hook.editor();
+
+    let changeEventCounter = 0;
+    let currentChangeEvent: { level: string; lastLevel: string };
+
+    const initialContent = '<p>some inital content</p>';
+    const manualModifiedLevel = 'a modified last level';
+
+    const assertChangeEvent = (event: { level: string; lastLevel: string }) => {
+      assert.equal(event.level, initialContent, 'Level has not the expected content');
+      assert.equal(event.lastLevel, manualModifiedLevel, 'Last level has not the expected content');
+    };
```

Comment:
Can we please put this outside of the test as we don't normally do inline assert functions like this for tests? For example at the top of this before any tests as thats where we normally put helper functions. It does mean you'll need to pass in the initial and modified content as variables though.

Reply:
Maybe it should be wrapped in a context and the function gets moved there. Since there will likely be more than one test here. I'm also thinking this might need to be split up it's testing a few different things if they where separate tests you would be able to describe what it's testing in each test.

Like `it('dispatchIfChanged should always fire on empty stack', ...)` or `it('dispatchIfChanged should compare with the level at the current undo index', ...)` for example.

Reply:
ok, I put a context and extracted the helper functions and variables